### ISSUE TITLE
$product->getRatingSummary() always returns null on products with ratings

### ISSUE THREAD

**Issue by ntoombs19** [magento/magento2#4530]
## Steps to reproduce
1. Override this template in a theme 'Magento_Catalog/templates/product/view/review.phtml`
2. Change the last parameter in the getReviewsSummaryHtml method to false so the method call in the template looks like this

``` php
<?php echo $block->getReviewsSummaryHtml($block->getProduct(), false, false)?>
```
## Expected result
1. A reviews summary will only be displayed for products that have ratings
## Actual result
1. A reviews summary is not displayed for any products; even ones that do have ratings.
## Additional Information
1. Set a break point on the first line of the `getReviewsSummary()` method in the class `\Magento\Review\Block\Product\ReviewRenderer`
2. Load the product detail page of a product that has a rating
3. Evaluate this expression once your breakpoint is hit `$product->getRatingSummary()`
4. It will always evaluate to null even with a product has a rating. Because of this, when $displayIfNoReviews is set to false, the if statement in which this line lives will evaluate to true and an empty string will be returned.

**Comment by oserediuk** [comment]
Hello @ntoombs19 ! I cannot reproduce this issue by your steps. Do you still have  this issue? Which code version do you use?

**Comment by vduglued** [comment]
I'm having the same issue. When the third parameter to `getReviewsSummaryHtml` is set to false, even if the product has reviews, the summary stars are not shown.

**Comment by vduglued** [comment]
I think @ntoombs19 made a mistake in his steps to reproduce. It should read:

> Change the last parameter in the getReviewsSummaryHtml method to **_false_**

The expectation is that when `$displayIfNoReviews` is `false`, the message "Be the first to write a review", is suppressed. When there _are_ reviews, the stars should display as normal.

The reality is that when `$displayIfNoReviews` is `false`, the message is suppressed, but so are the stars, even when the product has ratings.

The reason is, as @ntoombs19 points out, is that `$product->getRatingSummary()` always returns `null`.

In an ideal world, we'd also get rid of the boolean with the negative name, which would prevent the @ntoombs19 error and my three hundred edits of this comment for the sake of clarity and fixing my own mistakes ;)

**Comment by andimov** [comment]
@vduglued 
Thank you for clarification on this issue!
@ntoombs19 
Is it make sense for you?

**Comment by ntoombs19** [comment]
@andimov Yes that makes sense. I've corrected my mistake.
@vduglued Thank you for finding my error. It is indeed very confusing to have negative boolean variable names.

**Comment by andimov** [comment]
@ntoombs19 
Ok, thank you! So I'll close this issue as not a bug. Feel free to reopen if it needed.

**Comment by vduglued** [comment]
I think the bug is valid. The error was in reporting it in a slightly confusing way, but the bug is definitely real...

**Comment by ntoombs19** [comment]
@andimov I think you misunderstood. I made a mistake in my initial bug report saying to change the last parameter to true when I meant to say false. This is indeed a bug and it needs to be resolved. This issue needs to be reopened and my updated bug report needs to be reproduced and confirmed as a bug and then fixed.

**Comment by andimov** [comment]
@ntoombs19 @vduglued 
Ok!

**Comment by decius7bc** [comment]
"Hot fix" would be applying the following diff.
I.e. just loading the rating summary if it's "falsy" at the time of checking… There may be better options, though.

Will create a pull request eventually…

```diff
--- a/app/code/Magento/Review/Block/Product/ReviewRenderer.php
+++ b/app/code/Magento/Review/Block/Product/ReviewRenderer.php
@@ -57,6 +57,10 @@ class ReviewRenderer extends \Magento\Fr
         $templateType = self::DEFAULT_VIEW,
         $displayIfNoReviews = false
     ) {
+        if (!$product->getRatingSummary()) {
+            $this->_reviewFactory->create()->getEntitySummary($product, $this->_storeManager->getStore()->getId());
+        }
+
         if (!$product->getRatingSummary() && !$displayIfNoReviews) {
             return '';
         }
@@ -67,10 +71,6 @@ class ReviewRenderer extends \Magento\Fr
         $this->setTemplate($this->_availableTemplates[$templateType]);
 
         $this->setDisplayIfEmpty($displayIfNoReviews);
-
-        if (!$product->getRatingSummary()) {
-            $this->_reviewFactory->create()->getEntitySummary($product, $this->_storeManager->getStore()->getId());
-        }
         $this->setProduct($product);
 
         return $this->toHtml();
```

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-70758

**Comment by magento-engcom-team** [comment]
@ntoombs19, thank you for your report.
The issue is already fixed in develop branch
But we will consider to backport the fix to patch releases

**Comment by magento-engcom-team** [comment]
Hi @ntoombs19 the issue has been fixed in 2.2-develop branch and will be available with release 2.2.1 soon
